src/test/java/com/google/opensesame/projects/ProjectEntityTest.java

```diff
@@ -0,0 +1,161 @@
+
+import static com.googlecode.objectify.ObjectifyService.ofy;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.googlecode.objectify.ObjectifyFactory;
+import com.googlecode.objectify.ObjectifyService;
+import com.googlecode.objectify.util.Closeable;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.Collection;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ProjectEntityTest {
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private Closeable session;
+
+  private ProjectEntity firstMockProject;
+  private ProjectEntity secondMockProject;
+  private ProjectEntity thirdMockProject;
+
+  private HttpServletResponse mockResponse;
+  private StringWriter responseStringWriter;
+
+  @BeforeClass
+  public static void objectifyClassSetup() {
+    ObjectifyService.setFactory(new ObjectifyFactory());
+    ObjectifyService.register(ProjectEntity.class);
+  }
+
+  @Before
+  public void objectifyStartSession() throws IOException {
+    session = ObjectifyService.begin();
+    helper.setUp();
+  }
+
+  @Before
+  public void mockProjectSetUp() {
+    firstMockProject =
+        new ProjectEntity(
+            "273537467", Arrays.asList("Mentor 1", "Mentor 2", "Mentor 3"), Arrays.asList());
+    secondMockProject =
+        new ProjectEntity("45717250", Arrays.asList("Mentor 1", "Mentor 2"), Arrays.asList());
+    thirdMockProject = new ProjectEntity("20580498", Arrays.asList("Mentor 1"), Arrays.asList());
+    ofy().save().entities(firstMockProject, secondMockProject, thirdMockProject).now();
+  }
+
+  @Before
+  public void mockResponseSetUp() throws IOException {
+    mockResponse = mock(HttpServletResponse.class);
+    responseStringWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseStringWriter));
+  }
+
+  @After
+  public void objectifyEndSession() {
+    session.close();
+    helper.tearDown();
+  }
+
+  @Test
+  public void queryByIds() throws IOException {
```

Comment:
I think we can give the test more descriptive names, googling "test naming" or "Java test naming" gives lots of good examples.

I'm used to tests having names like `testqueryFromRequestReturnsCorrectProjects` for example. So I can read the test name and understand what it's supposed to do, without reading the comment.

Reply:
Okay good to know. I'll do a pass over all of my tests with this in mind!